### PR TITLE
Docs: components updates

### PR DESCRIPTION
**Pull request by mendonk**
Add a "generic" component usage section at the top of each component page, with additional pointers or gotchas to help users use the component of their choice.

Some additional fixes:
1. Move custom components page out of helpers.
2. Remove and redirect RAG component page, out of date.
3. Move components to below workspace.
4. Processing page up to date.
5. Tip for loaders page which are now in Bundles in the UI.

### CODE REVIEW

File: docs/docs/Components/components-vector-stores.md [hunk 1]
```diff
@@ -559,20 +544,45 @@ For more information, see the [Vectara documentation](https://docs.vectara.com/d
...
+| vectara_corpus_id     | String       | Vectara corpus ID                                          |
+| vectara_api_key       | SecretString | Vectara API key                                            |
+| search_query          | String       | The query to receive an answer on                          |
+| lexical_interpolation | Float        | Hybrid search factor (0.005 to 0.1)                        |
+| filter                | String       | Metadata filters to narrow the search                      |
+| reranker              | String       | Reranker type (mmr, rerank_multilingual_v1, none)          |
+| reranker_k            | Integer      | Number of results to rerank (1 to 100)                     |
+| diversity_bias        | Float        | Diversity bias for MMR reranker (0 to 1)                   |
+| max_results           | Integer      | Maximum number of search results to summarize (1 to 100)   |
+| response_lang         | String       | Language code for the response (e.g., "eng", "auto")       |
```
Comment (KimberlyFields) [inline docs/docs/Components/components-vector-stores.md:571]: Per the Google style guide, avoid "e.g.". Use "for example" instead.

File: docs/docs/Components/components-data.md [hunk 2]
```diff
@@ -4,6 +4,23 @@ sidebar_position: 3
 slug: /components-data
 ---
 
+Data components load data from a source into your flow.
```
[the remark this answers is not in the document]
Comment (mendonk) [reply docs/docs/Components/components-data.md:9]: Updated this for all pages

File: docs/docs/Components/components-helpers.md [hunk 3]
```diff
@@ -4,233 +4,90 @@ sidebar_position: 4
...
 | Name | Display Name | Info |
 |------|--------------|------|
-| texts | Texts | Texts to split. |
-| separators | Separators | Characters to split on. Defaults to a space. |
-| max_chunk_size | Max Chunk Size | The maximum length (in characters) of each chunk. |
-| chunk_overlap | Chunk Overlap | The amount of character overlap between chunks. |
-| recursive | Recursive | Whether to split recursively. |
+| messages | Messages (Data) | Retrieved messages as Data objects. |
+| messages_text | Messages (Text) | Retrieved messages formatted as text. |
+| lc_memory | Memory | Built LangChain memory object. |
```
[the remark this answers is not in the document]
Comment (mendonk) [reply docs/docs/Components/components-helpers.md]: Added some context - it is a Langchain ConversationBufferMemory object, and linked out to their docs

[the remark this answers is not in the document]
Comment (mendonk) [reply docs/docs/Components/components-helpers.md]: *LLMs

File: docs/docs/Components/components-memories.md [hunk 4]
```diff
@@ -1,14 +1,24 @@
 # Memories
 
-Chat memory components store and retrieve chat messages by `session_id`.
+Memory components store and retrieve chat messages by `session_id`.
 
-## AstraDBChatMemory Component
+They are distinct from vector store components, because they are built specifically for storing and retrieving chat messages from external databases.
+
+Memory components provide access to their respective external databases **as memory**. This allows AIs to access external memory for persistence and context retention.
```
[the remark this answers is not in the document]
Comment (mendonk) [reply docs/docs/Components/components-memories.md:12]: * LLMs

File: docs/docs/Components/components-vector-stores.md [hunk 5]
```diff
@@ -1,21 +1,41 @@
...
-# Vector Stores
 
-Vector databases are used to store and search for vectors. They can be used to store embeddings, search for similar vectors, and perform other vector operations.
+# Vector store components in Langflow
+
+Vector databases store vector data, which backs AI workloads like chatbots and Retrieval Augmented Generation.
+
+Vector database components establish connections to existing vector databases or create in-memory vector stores for storing and retrieving vector data.
+
+Vector database components are distinct from [memory components](/components-memories), which are are built specifically for storing and retrieving chat messages from external databases.
```
Comment (KimberlyFields) [inline docs/docs/Components/components-vector-stores.md:12]:
```suggestion
Vector database components are distinct from [memory components](/components-memories), which are built specifically for storing and retrieving chat messages from external databases.
```

File: docs/docs/Components/components-vector-stores.md [hunk 6]
```diff
@@ -1,21 +1,41 @@
...
+
+## Use a vector store component in a flow
+
+Vector databases can be populated from within Langflow with document ingestion pipelines, like this one.
```
Comment (KimberlyFields) [inline docs/docs/Components/components-vector-stores.md:16]:
```suggestion
Vector databases can be populated from within Langflow with document ingestion pipelines, like the following.
```

**Review by KimberlyFields — APPROVED** [review]
I left a few comments and suggestions, but I'm approving.

File: docs/docs/Components/components-helpers.md [hunk 7]
```diff
@@ -4,233 +4,92 @@ sidebar_position: 4
...
-| Name | Display Name | Info |
-|------|--------------|------|
-| filtered_data | Filtered Data | Filtered data as a list of Data objects. |
-
-## Merge Data
+## Message history
 
-This component combines multiple data sources into a single unified Data object.
+:::info
+Prior to Langflow 1.1, this component was known as the Chat Memory component.
```
[the remark this answers is not in the document]
Comment (mendonk) [reply docs/docs/Components/components-helpers.md:61]: Added a notion ticket also

File: docs/docs/Components/components-overview.md [hunk 8]
```diff
@@ -1,71 +1,106 @@
...
 
-A component is a single building block within a flow. It consists of inputs, outputs, and parameters that define their functionality. These elements provide a convenient and  straightforward way to compose LLM-based applications. Learn more about components and how they work below.
+import Icon from "@site/src/components/icon";
 
+A component is a single building block within a flow with inputs, outputs, functions, and parameters that define its functionality. A single component is like a class within a larger application.
 
-During the flow creation process, you will notice handles (colored circles) attached to one or both sides of a component. These handles use distinct colors to indicate the types of inputs and outputs that can be interconnected. Hover over a handle to see connection details.
+To add a component to a flow, drag it from the **Component** menu to the **Workspace**.
 
+Learn more about components and how they work below.
```
[the remark this answers is not in the document]
Comment (mendonk) [reply docs/docs/Components/components-overview.md]: Fixed

**Pull request merged**